packages/url/src/clean-for-slug.js

```diff
@@ -24,7 +24,7 @@ export function cleanForSlug( string ) {
 		return '';
 	}
 	return (
-		removeAccents( string )
+		removeAccents( string.replace( /&amp;/g, '' ).replace( /&lt;/g, '' ) )
```

Comment:
What about any other commonly used HTML entities, like `&gt;`, `&nbsp;`, `&apos;`, `&quot;` and others (not an exhaustive list by any means)?

Reply:
@tyxla Yes, I agree with you, it should handle all entities. I wavered between keeping it limited to what you would get from the WP REST API response for the title (only those entities need to be handled) and all entities, wasn't quite sure which approach to take!

Reply:
I see. In light of the recent conversation to stay compatible with the `sanitize_title_with_dashes` filter, let's introduce a utility function compatible with `sanitize_title` filtered by `sanitize_title_with_dashes` which converts some entities (spaces and dashes) to hyphens, and obliterates the rest of the HTML entities.